packages/library/src/utils/fetchDrawIdToPush.ts

```diff
@@ -0,0 +1,55 @@
+import { Contract } from '@ethersproject/contracts';
+import { Draw } from '../types';
+
+/**
+ * Fetches the latest draw id from the draw buffer and increments it by 1.
+ * PoolTogether v4 v1.3.0 Architecture
+ * @param drawBuffer
+ * @param prizeDistributionBuffer
+ * @returns
+ */
+export async function fetchDrawIdToPush(drawBuffer: Contract, prizeDistributionBuffer: Contract) {
+  let newestDraw: Draw;
+  let drawIdToPush = 0;
+  let newestPrizeDistributionDrawId = 0;
+  let shouldPush = false;
+
+  /**
+   * Fetch newest Draw
+   * IF the fetch throws an error, the DrawBuffer is not initialized
+   */
+  try {
+    newestDraw = await drawBuffer.getNewestDraw();
```

Comment:
This file is pretty similar to the `calculateDrawToPushToTimelock.ts` one.
https://github.com/pooltogether/v4-autotasks/blob/369bfd1bb089008329ce9098b8f2d8f1fcbe24e1/packages/library/src/utils/calculateDrawToPushToTimelock.ts#L34

So we can call the function `fetchDrawIdToPush` in `calculateDrawToPushToTimelock.ts`.

Reply:
I've kept `calculateDrawToPushToTimelock` for backwards compatibility. If we're keeping it then there's no reason to edit the legacy autotasks.